test/e2e/windows/memory_limits.go

```diff
@@ -105,31 +104,66 @@ func overrideAllocatableMemoryTest(f *framework.Framework, allocatablePods int)
 		podType = "memory_limit_test_pod"
 	)
 
-	totalAllocatable := getTotalAllocatableMemory(f)
-
-	memValue := totalAllocatable.Value()
-	memPerPod := memValue / int64(allocatablePods)
-	ginkgo.By(fmt.Sprintf("Deploying %d pods with mem limit %v, then one additional pod", allocatablePods, memPerPod))
-
-	// these should all work
-	pods := newMemLimitTestPods(allocatablePods, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().CreateBatch(pods)
-
-	failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().Create(failurePods[0])
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	framework.ExpectNoError(err)
 
+	for _, node := range nodeList.Items {
+		status := node.Status
+		pod := v1.Pod{
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Image: imageutils.GetPauseImageName(),
+						Resources: v1.ResourceRequirements{
+							Limits: v1.ResourceList{
+								v1.ResourceMemory: status.Allocatable[v1.ResourceMemory],
+							},
+						},
+					},
+				},
+				NodeSelector: map[string]string{
+					"kubernetes.io/os": "windows",
+				},
+			},
+		}
+		pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), &pod, metav1.CreateOptions{})
```

Comment:
@jsturtevant I think `&pod` should be correct argument for this case and `&failurePod` for the next case but it is showing an error. It is saying that pointer type should not be used but in [this](https://github.com/kubernetes/kubernetes/blob/7dd03c8326858ceb232d38476db3240a15e8494f/test/e2e/windows/dns.go#L61-L62) example pointer to `v1.Pod` type is used in the argument.

What is the cause of the error here ?

Reply:
Tried with only `pod` and `failurePod` too, but still it is showing an error.

Reply:
@jsturtevant Could you please give your thoughts on the above comment ?

Reply:
@jsturtevant Could you please give your thoughts on the above comment ?

Reply:
```suggestion
		pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
```

Reply:
This with the change https://github.com/kubernetes/kubernetes/pull/107477/files#r798791328 should work